tests/virt_autotest/qem/live_migration_src.pm

```diff
@@ -0,0 +1,217 @@
+        validate_script_output("curl -s -k -u bernhard:$testapi::password scp://$vm_info->{ip}/~/test_vm_write", sub { m/$vm_name/ });
+        assert_script_run("rm $images_path/guestip/$vm_name");
+        #shutdown migrated vm
+        if ($cmd eq "xl migrate") {
+            assert_script_run("ssh root\@$dst_ip xl shutdown $vm_name");
+        } else {
+            assert_script_run("ssh root\@$dst_ip virsh shutdown $vm_name");
+        }
+        #wait for shutdown complete
+        sleep 15;
+        #start again for next migration test
+        assert_script_run("virsh start $vm_name") if ($cmd ne $vm_info->{virsh_cmds}->[-1]);
+    }
+}
+
+sub cleanup {
+    my ($vm_name, $vm_info) = @_;
+    assert_script_run("virsh dumpxml $vm_name > $images_path/vm-configs/$vm_name.xml") unless ($vm_info->{reuse});
+    assert_script_run("virsh undefine $vm_name");
+}
+
+sub run {
+    select_console 'root-ssh';
+    $dst_ip = get_job_autoinst_vars(get_parents->[0])->{SUT_IP};
+    record_info("DST: $dst_ip");
+    record_info("SUT: $sut_ip");
+    assert_script_run("mount $sut_ip:/tmp/vm_images $images_path") unless get_var("NFS_PATH");
+    assert_script_run("rm -rf $images_path/guestip; mkdir $images_path/guestip");
+    assert_script_run("mkdir -p $images_path/vm-configs");
+    my $guests = get_var_array('GUESTS');
+    #start all guests
+    for my $guest (@{$guests}) { start_guest($guest, $vms); }
+
+    assert_script_run('if [ -d /root/.ssh ]; then rm -rf /root/.ssh; fi');
+    assert_script_run('ssh-keygen -t rsa -q -f "/root/.ssh/id_rsa" -N ""');
+
+    mutex_wait 'dst_ready';
+
+    assert_script_run("ssh-keyscan $dst_ip > /root/.ssh/known_hosts");
+    assert_script_run("curl -s -k --user root:$testapi::password -T .ssh/id_rsa.pub sftp://$dst_ip/root/.ssh/authorized_keys  --ftp-create-dirs");
+    assert_script_run("ssh root\@$dst_ip mount $sut_ip:/tmp/vm_images $images_path") unless get_var("NFS_PATH");
+
+    my $n         = keys %$vms;
+    my $time_cont = 0;
+    #Check each vm if their have got IP address. If they have, their install or stratup are supposed to complete and ready for migration test.
+    until ($n == 0) {
+        while (my ($vm_name, $vm_info) = each %$vms) {
+            next if ($vm_info->{finish} == 1);
+            #check if vm write its IP address in $images_path/guestip/$vm_name on src host.
+            if (script_run("[[ -f $images_path/guestip/$vm_name ]]") == 0) {
+                record_info("$vm_name: ready");
+                migrate_test($vm_name, $vm_info);
+                cleanup($vm_name, $vm_info);
+                $vm_info->{finish} = 1;
+                $n--;
+            } else {
+                die("$vm_name startup time out after " . $time_cont * 2 . " minutes") if ($time_cont == 20);
+                my $vm_state = script_output("virsh domstate --domain $vm_name");
+                if ($vm_state eq "shut off") {
```

Comment:
Is this enough to tell vm creation is successful? There can be various failures but vm is shutoff.

Reply:
Here is just to start vm if a vm is found in "shut off" state. 
We use /etc/init.d/after.local in vm to verify  if a vm creation and installation is successful.  after.local will be executed after boot and services startup.  after.local will mount the "shared storage", create a file with it vm name and write its IP address in the file, then unmount it.
This automation will check whether this "IP file"  is present or not.  If  present we assume this vm is working.  You can find the after.local code in guest profile in "Logs & Assets".

Reply:
We use "after.local" to determine if boot process is done or not. 
Guests here all reboot twice by autoyast during installation.  after.local will be called by the last boot after all services startup.  We can assume that the booting process is done after after.local is invoked. That what after.local is meant for.